From: David Kimdon <dwhedon@debian.org>
Date: Tue, 10 Feb 2004 15:58:12 -0600
Subject: Make default hook templates convenient to use

Newly created repositories have a post-commit.tmpl that calls
commit-email.pl and then log-commit.py. It does not first set a path,
so unless someone edits the hook, commit-email.pl cannot be found at
all.  Likewise for other scripts the hook calls.

[peter@p12n.org, 2005-12-03: Now hooks live in /usr/share/subversion,
 not /usr/lib/subversion.]

[peter@p12n.org, 2009-03-20: Change /usr/share/subversion/hook-scripts/
 to $REPOS/hooks/ everywhere in the hook script .tmpl example files.  I
 want to encourage repository administrators to copy the hooks, not just
 use them in place.  Makes it easier to upgrade without breaking stuff.]

Fixes: http://bugs.debian.org/210901
 
Index: subversion-1.7.0/subversion/libsvn_repos/repos.c
===================================================================
--- subversion-1.7.0.orig/subversion/libsvn_repos/repos.c	2011-10-20 20:47:32.000000000 +0200
+++ subversion-1.7.0/subversion/libsvn_repos/repos.c	2011-10-20 21:55:48.000000000 +0200
@@ -275,7 +275,7 @@
 
 #define PREWRITTEN_HOOKS_TEXT                                                 \
   "# For more examples and pre-written hooks, see those in"                NL \
-  "# the Subversion repository at"                                         NL \
+  "# /usr/share/subversion/hook-scripts, and in the repository at"         NL \
   "# http://svn.apache.org/repos/asf/subversion/trunk/tools/hook-scripts/ and"        NL \
   "# http://svn.apache.org/repos/asf/subversion/trunk/contrib/hook-scripts/"          NL
 
@@ -354,8 +354,11 @@
 "REPOS=\"$1\""                                                               NL
 "USER=\"$2\""                                                                NL
 ""                                                                           NL
-"commit-allower.pl --repository \"$REPOS\" --user \"$USER\" || exit 1"       NL
-"special-auth-check.py --user \"$USER\" --auth-level 3 || exit 1"            NL
+"# Exit on all errors."                                                      NL
+"set -e"                                                                     NL
+""                                                                           NL
+"\"$REPOS\"/hooks/commit-allower.pl --repository \"$REPOS\" --user \"$USER\"" NL
+"\"$REPOS\"/hooks/special-auth-check.py --user \"$USER\" --auth-level 3"     NL
 ""                                                                           NL
 "# All checks passed, so allow the commit."                                  NL
 "exit 0"                                                                     NL;
@@ -442,10 +445,13 @@
 "$SVNLOOK log -t \"$TXN\" \"$REPOS\" | \\"                                   NL
 "   grep \"[a-zA-Z0-9]\" > /dev/null || exit 1"                              NL
 ""                                                                           NL
+"# Exit on all errors."                                                      NL
+"set -e"                                                                     NL
+""                                                                           NL
 "# Check that the author of this commit has the rights to perform"           NL
 "# the commit on the files and directories being modified."                  NL
-"commit-access-control.pl \"$REPOS\" \"$TXN\" commit-access-control.cfg || exit 1"
-                                                                             NL
+"\"$REPOS\"/hooks/commit-access-control.pl \"$REPOS\" $TXN \\"               NL
+"  \"$REPOS\"/hooks/commit-access-control.cfg"                               NL
 ""                                                                           NL
 "# All checks passed, so allow the commit."                                  NL
 "exit 0"                                                                     NL;
@@ -756,7 +762,7 @@
 "REPOS=\"$1\""                                                               NL
 "REV=\"$2\""                                                                 NL
                                                                              NL
-"mailer.py commit \"$REPOS\" \"$REV\" /path/to/mailer.conf"                  NL;
+"\"$REPOS\"/hooks/mailer.py commit \"$REPOS\" $REV \"$REPOS\"/mailer.conf"   NL;
 
 #undef SCRIPT_NAME
 
@@ -817,7 +823,8 @@
 "USER=\"$2\""                                                                NL
 ""                                                                           NL
 "# Send email to interested parties, let them know a lock was created:"      NL
-"mailer.py lock \"$REPOS\" \"$USER\" /path/to/mailer.conf"                   NL;
+"\"$REPOS\"/hooks/mailer.py lock \\"                                         NL
+"  \"$REPOS\" \"$USER\" \"$REPOS\"/hooks/mailer.conf"                        NL;
 
 #undef SCRIPT_NAME
 
@@ -876,7 +883,8 @@
 "USER=\"$2\""                                                                NL
 ""                                                                           NL
 "# Send email to interested parties, let them know a lock was removed:"      NL
-"mailer.py unlock \"$REPOS\" \"$USER\" /path/to/mailer.conf"                 NL;
+"\"$REPOS\"/hooks/mailer.py unlock \\"                                       NL
+"  \"$REPOS\" \"$USER\" \"$REPOS\"/hooks/mailer.conf"                        NL;
 
 #undef SCRIPT_NAME
 
@@ -941,8 +949,8 @@
 "PROPNAME=\"$4\""                                                            NL
 "ACTION=\"$5\""                                                              NL
 ""                                                                           NL
-"mailer.py propchange2 \"$REPOS\" \"$REV\" \"$USER\" \"$PROPNAME\" "
-"\"$ACTION\" /path/to/mailer.conf"                                           NL;
+"\"$REPOS\"/hooks/mailer.py propchange2 \"$REPOS\" $REV \\"                  NL
+"  \"$USER\" \"$PROPNAME\" \"$ACTION\" \"$REPOS\"/hooks/mailer.conf"         NL;
 
 #undef SCRIPT_NAME
 
